File: Csharp/RT_CS/RT_CS/Materials/ColorMaterial.cs
using RT_CS.shapes;
using RT_CS.utility;
using System;
using System.Collections.Generic;
using System.Text;

namespace RT_CS.Materials
{
    public class ColorMaterial : Material
    {
        private List<Vector3> ambientColors, diffuseColors, specularColors;
   
        public ColorMaterial(int mid, List<float> spcs, MaterialType mt, List<float> refl, List<float> trans, List<float> refr, List<Vector3> ac,List<Vector3> dc, List<Vector3> sc) : base(mid,spcs,mt,refl,trans,refr)
        {
            ambientColors = ac;
            diffuseColors = dc;
            specularColors = sc;
        }
        public override Vector3 GetAmbientColor(int sid, Vector3 point)
        {
            return ambientColors[0];
        }

        public override Vector3 GetDiffuseColor(int sid, Vector3 point)
        {
            return diffuseColors[0];
        }

        public override Vector3 GetSpecularColor(int sid, Vector3 point)
        {
            return specularColors[0];
        }

        public override float GetSpecularity(int sid, Vector3 point)
        {
            return specularities[0];
        }

        public override float GetReflectivity(int sid, Vector3 point)
        {
            return reflectivity[0];
        }
        public override float GetTransparency(int sid, Vector3 point)
        {
            return transparency[0];
        }
        public override float GetRefractiveIndex(int sid, Vector3 point)
        {
            return refractiveIndices[0];    
        }
    }
}

File: cpp/RT_CPP/RT_CPP/color_material.cpp
#include "color_material.h"

using namespace materials;

color_material::color_material(int mid, material_types tp, vector<vector3> acs,
	vector<vector3> dcs, vector<vector3> spcs, vector<float> specs,
	vector<float> rfs, vector<float> rts, vector<float> rfIs) : material(mid,tp,specs,rfs,rts, rfIs),
	ambient_colors(acs),diffuse_colors(dcs),specular_colors(spcs)
{

}

vector3 color_material::get_ambient_color(vector2 uv) const
{
	return ambient_colors[0];
}

vector3 color_material::get_diffuse_color(vector2 uv) const
{
	return diffuse_colors[0];
}
vector3 color_material::get_specular_color(vector2 uv) const
{
	return specular_colors[0];
}
float color_material::get_specularity(vector2 uv) const
{
	return specularities[0];
}

float color_material::get_reflectivity(vector2 uv) const
{
	return reflectivities[0];
}

float color_material::get_transparency(vector2 uv) const
{
	return transparencies[0];
}

float color_material::get_refractive_index(vector2 uv) const
{
	return refractiveIndices[0];
}